### PR TITLE
Handle NaN

### PR DESCRIPTION
**Pull request by valichek**
Fixes: https://github.com/mcollina/msgpack5/issues/74.

### CODE REVIEW

File: test/15-elements-maps.js [hunk 1]
```diff
@@ -71,6 +71,16 @@ test('do not encode undefined in a map', function (t) {
   t.end()
 })
 
+test('do not encode NaN in a map', function (t) {
+  var instance = msgpack()
+  var expected = { hello: 'world' }
+  var toEncode = { a: NaN, hello: 'world' }
+  var buf = instance.encode(toEncode)
+
+  t.deepEqual(expected, instance.decode(buf), 'must ignore NaN')
```
Comment (mcollina) [inline test/15-elements-maps.js]: why? Shouldn't it have the same behavior as encoded directly?

Comment (valichek) [reply test/15-elements-maps.js]: did it same as for `undefined` value

Comment (mcollina) [reply test/15-elements-maps.js]: They are different. The behavior of `undefined`  is matching how JSON.stringify() works. `undefined`  in javascript means "there is nothing here". `NaN`  has a different meaning altogether. It should throw anyway.

Comment (valichek) [reply test/15-elements-maps.js]: Ok, will remove that

**Pull request merged**